src/main/java/org/openstreetmap/atlas/generator/tools/spark/persistence/PersistenceTools.java

```diff
@@ -43,8 +46,24 @@ public PersistenceTools(final Map<String, String> configurationMap)
 
     public CountryBoundaryMap boundaries(final String input)
     {
-        return CountryBoundaryMap.fromPlainText(SparkJob
-                .resource(SparkFileHelper.combine(input, BOUNDARIES_FILE), this.configurationMap));
+        try (ResourceCloseable resource = SparkJob
+                .resource(SparkFileHelper.combine(input, BOUNDARIES_FILE), this.configurationMap))
+        {
+            return CountryBoundaryMap.fromPlainText(resource);
+        }
+        catch (final Exception e)
+        {
+            logger.error(e);
+            if (e instanceof RuntimeException)
+            {
+                throw (RuntimeException) e;
+            }
+            else
+            {
+                throw new CoreException("Could not close {}", e,
+                        SparkFileHelper.combine(input, BOUNDARIES_FILE));
```

Comment:
```suggestion
                throw new CoreException("Could not close {}",
                        SparkFileHelper.combine(input, BOUNDARIES_FILE), e);
```

Reply:
I see how this works. Is it a good idea to do that though? The `CoreException` constructor for that is undocumented, so the behavior that uses the _last_ argument as the cause could change at any time. I'd rather use the explicit constructor where I know that the `Throwable` is in the appropriate location. If this change is truly desired, then maybe docs should indicate that the `CoreException(String, Object...)` is preferred over `CoreException(String, Throwable, Object...)`. I'll open up a PR for that if needed.

Reply:
Ping.